### ISSUE TITLE
Output JPEG images can not be decoded by CyberLink PhotoDirector

### ISSUE THREAD

**Issue by cdfmr** [ImageOptim/ImageOptim#172]
![imageoptim](https://cloud.githubusercontent.com/assets/1453996/20252031/7df2045e-aa59-11e6-99b0-710bfcdcfc14.jpg)

The left is what I see in Preview.app, and the right is what I get in PhotoDirector 7/8.

**Comment by curby** [comment]
Can you upload or link to an image that's not rendering correctly?  I'm not a dev but I'm curious what my tools and programs do when they open a problem image.

**Comment by pornel** [comment]
You need to write to PhotoDirector and ask them to fix their JPEG support.

**Comment by cdfmr** [comment]
@pornel I'm contacting CyberLink, but now I found these images can not be opened by Nikon ViewNX-i too.
@curby Just use ImageOptim to process a JPEG file, maybe it's a compatibility bug of PhotoDirector and ViewNX.

**Comment by pornel** [comment]
ImageOptim makes JPEGs smaller by using some less common features of the format (custom progressive scan scripts). These are fully supported by libjpeg which powers 90% of desktop software, but custom primitive/buggy JPEG decoders may have problems with that.

I can't do much about that. If I stick to baseline JPEG that primitive decoders support, then optimization will do very little. Basically if you need compatibility with a tool that chokes on progressive JPEG then you can't use ImageOptim.

**Comment by cdfmr** [comment]
@pornel Thank you for your help, now I know what the problem is. Unfortunately I've processed thousands of my pictures with ImageOptim and deleted original ones, is there a way to convert them back without loss quality?

**Comment by pornel** [comment]
Yes. If you haven't emptied Trash yet, original files should be there.

You can change progressive to baseline with this shell command:

```sh
/Applications/ImageOptim.app/Contents/Frameworks/ImageOptimGPL.framework/Resources/jpegtran -copy all -fastcrush -outfile converted.jpg original.jpg
```

this assumes there's `original.jpg` in the current directory and it will (over)write `converted.jpg`.

**Comment by cdfmr** [comment]
Thank you very much, I'll try it.
